presto-kinesis/src/main/java/io/prestosql/plugin/kinesis/s3config/S3TableConfigClient.java

```diff
@@ -0,0 +1,266 @@
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.prestosql.plugin.kinesis.s3config;
+
+import com.amazonaws.AmazonClientException;
+import com.amazonaws.AmazonServiceException;
+import com.amazonaws.services.s3.AmazonS3Client;
+import com.amazonaws.services.s3.AmazonS3URI;
+import com.amazonaws.services.s3.model.GetObjectRequest;
+import com.amazonaws.services.s3.model.ListObjectsRequest;
+import com.amazonaws.services.s3.model.ObjectListing;
+import com.amazonaws.services.s3.model.S3Object;
+import com.amazonaws.services.s3.model.S3ObjectSummary;
+import com.google.common.collect.ImmutableMap;
+import com.google.inject.Inject;
+import io.airlift.json.JsonCodec;
+import io.airlift.log.Logger;
+import io.prestosql.plugin.kinesis.ConnectorShutdown;
+import io.prestosql.plugin.kinesis.KinesisClientProvider;
+import io.prestosql.plugin.kinesis.KinesisConnectorConfig;
+import io.prestosql.plugin.kinesis.KinesisStreamDescription;
+import io.prestosql.spi.connector.SchemaTableName;
+
+import java.io.BufferedReader;
+import java.io.IOException;
+import java.io.InputStreamReader;
+import java.util.ArrayList;
+import java.util.Collection;
+import java.util.Collections;
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.Executors;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.ScheduledFuture;
+import java.util.concurrent.TimeUnit;
+
+import static java.util.Objects.requireNonNull;
+
+/**
+ * Utility class to retrieve table definitions from a common place on Amazon S3.
+ * <p>
+ * This is so that we can add new tables in a central "metastore" location without
+ * having to update every single node with the files.
+ * <p>
+ * This makes calls to Amazon AWS using the S3 client.
+ */
+public class S3TableConfigClient
+        implements ConnectorShutdown
+{
+    private static final Logger log = Logger.get(S3TableConfigClient.class);
+
+    public final KinesisConnectorConfig kinesisConnectorConfig;
+    private final KinesisClientProvider clientManager;
+    private final JsonCodec<KinesisStreamDescription> streamDescriptionCodec;
+
+    private final String bucketUrl;
+    private long lastCheck;
+    private ScheduledFuture<?> updateTaskHandle;
+
+    private Map<String, KinesisStreamDescription> internalMap =
+            Collections.synchronizedMap(new HashMap<String, KinesisStreamDescription>());
+
+    @Inject
+    public S3TableConfigClient(KinesisConnectorConfig aConnectorConfig,
+            KinesisClientProvider aClientManager,
```

Comment:
Don't prefix parameters with `a`. The ambiguity with field names in constructors is resolved by using `this.` for the field names.

Reply:
changed to proper format